tests/test_script_queue.py

```diff
@@ -245,15 +245,18 @@ def test_invalid_paths(self):
 
 
 class ScriptQueueTestCase(salobj.BaseCscTestCase, unittest.IsolatedAsyncioTestCase):
-    def basic_make_csc(self, initial_state, config_dir=None, simulation_mode=0):
+    @classmethod
+    def setUpClass(cls):
         datadir = os.path.abspath(os.path.join(os.path.dirname(__file__), "data"))
-        standardpath = os.path.join(datadir, "standard")
-        externalpath = os.path.join(datadir, "external")
+        cls.standardpath = os.path.join(datadir, "standard")
+        cls.externalpath = os.path.join(datadir, "external")
```

Comment:
Please move this to the existing `setUp` method. It's cheap enough to compute every time fresh, and `self.datadir` is already being set there. Please reserve `setUpClass` for operations that are slow and expensive enough to only want to do them once.

Note: my existing code should have been using self.datadir instead of recomputing it in basic_make_csc.